x/observer/simulation/operations.go

```diff
@@ -23,85 +27,389 @@ import (
 // Based on the weights assigned in the cosmos sdk modules,
 // 100 seems to the max weight used,and we should use relative weights
 // to signify the number of each operation in a block.
-
-// TODO Add more details to comment based on what the number represents in terms of percentage of operations in a block
-// https://github.com/zeta-chain/node/issues/3100
 const (
-	// #nosec G101 not a hardcoded credential
-	OpWeightMsgTypeMsgEnableCCTX = "op_weight_msg_enable_crosschain_flags"
-	// DefaultWeightMsgTypeMsgEnableCCTX We ues a high weight for this operation
-	// to ensure that it is present in the block more number of times than any operation that changes the validator set
+	OpWeightMsgTypeMsgEnableCCTX                  = "op_weight_msg_enable_crosschain_flags"         // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgDisableCCTX                 = "op_weight_msg_disable_crosschain_flags"        // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgVoteTSS                     = "op_weight_msg_vote_tss"                        // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgUpdateKeygen                = "op_weight_msg_update_keygen"                   // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgUpdateObserver              = "op_weight_msg_update_observer"                 // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgUpdateChainParams           = "op_weight_msg_update_chain_params"             // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgRemoveChainParams           = "op_weight_msg_remove_chain_params"             // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgResetChainNonces            = "op_weight_msg_reset_chain_nonces"              // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgUpdateGasPriceIncreaseFlags = "op_weight_msg_update_gas_price_increase_flags" // #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgAddObserver                 = "op_weight_msg_add_observer"                    // #nosec G101 not a hardcoded credential
 
+	// DefaultWeightMsgTypeMsgEnableCCTX We use a high weight for this operation
+	// to ensure that it is present in the block more number of times than any operation that changes the validator set
 	// Arrived at this number based on the weights used in the cosmos sdk staking module and through some trial and error
-	DefaultWeightMsgTypeMsgEnableCCTX = 3650
+	DefaultWeightMsgTypeMsgEnableCCTX                  = 3650
```

Comment:
Previously you mention
> Based on the weights assigned in the cosmos sdk modules , 100 seems to the max weight used

Reply:
That is correct; we want to make sure that this message is included in blocks more times than any message that modifies the observer set. 
So, more than the combined number for any msg that changes the staking amount, removes the validator etc.


The reason we need it so , is becuase any of those changes automatically disable inbound transactions, and this is the only message to enable it back on.